runtime/src/bank.rs

```diff
@@ -1484,14 +1484,16 @@ impl Bank {
             );
 
             let post_balance = self.deposit(&self.collector_id, unburned);
-            self.rewards.write().unwrap().push((
-                self.collector_id,
-                RewardInfo {
-                    reward_type: RewardType::Fee,
-                    lamports: unburned as i64,
-                    post_balance,
-                },
-            ));
+            if unburned != 0 {
+                self.rewards.write().unwrap().push((
```

Comment:
The change of behavior shouldn't cause any problem for cluster consensus.

But how about defining a method like `fn {record_positive_reward|record_reward_if_any}(&mut rewards, reward_type, pubkey, lamports, post_balance)`? The behavior of skipping to push for 0-lamport rewards seems quite generic and solid invariant tied to `Bank::rewards`. The same code is repeated thrice in `collect_fees`, `pay_validator_rewards` and  `distribute_rent_to_validators`.

Reply:
Nice idea. I think that trying to create that generic function will not work out well, each is slightly different so I think I'd rather just leave it as is.